移出model和dataload

xiangyuli-bio opened the pull request:
1.baselint_datamodule.py 添加缓存和移出分段逻辑 合并TDCdataload的功能
2.dg_datamodule.py 作为基类 添加bins和classify属性
3.morgan_chembert_datmodule.py 继承baseline_datamodule.py的类
  实现同时加载morgan和chembert的功能 合并了TDCdataload_double的功能
4.base_model.py 完善一些细节 加入了CLM:loss和predict
5.morgan_model.py 实现了原本DrugAttentionmodel的功能
6.morgan_chembert_model,py 继承morganmodel
  实现了DrugAttnetionmodel_Double的功能 用于morgan
  chembert同时加载的模型
7.train.py 对于morgan_model
  需要传入drug_shape由于morgan还是chembert的不同，加入Ensemble_Learn参数来选择分类头个数

xiangyuli-bio committed:
解决val损失问题

1.architectures.py和base_model.py
  移除architectures.py中验证时的ordinal_regression_predict函数到base_model.py
2.在morgan_model.py中调整逻辑 先计算损失 再转化预测one-hot

xiangyuli-bio committed:
移出model和dataload

1.baselint_datamodule.py 添加缓存和移出分段逻辑 合并TDCdataload的功能
2.dg_datamodule.py 作为基类 添加bins和classify属性
3.morgan_chembert_datmodule.py 继承baseline_datamodule.py的类
  实现同时加载morgan和chembert的功能 合并了TDCdataload_double的功能
4.base_model.py 完善一些细节 加入了CLM:loss和predict
5.morgan_model.py 实现了原本DrugAttentionmodel的功能
6.morgan_chembert_model,py 继承morganmodel
  实现了DrugAttnetionmodel_Double的功能 用于morgan
  chembert同时加载的模型
7.train.py 对于morgan_model
  需要传入drug_shape由于morgan还是chembert的不同，加入Ensemble_Learn参数来选择分类头个数

fireinice commented on src/datamodule/morgan_chembert_datamodule.py:
```diff
@@ -2,56 +2,181 @@
+
+        self.drug_featurizer_two = self.drug_featurizer[1]
+        self.drug_featurizer = self.drug_featurizer[0]
+    def prepare_data(self):
```
这里没有行分隔

fireinice commented on src/datamodule/morgan_chembert_datamodule.py:
```diff
@@ -2,56 +2,181 @@
+        self.prepare_featurizer(self.drug_featurizer_two,self.all_drugs)
+
+    def setup(self, stage: str):
+        self.setup_featurizer(self.target_featurizer, self.all_targets)
```
这里有比较多的代码重复，考虑单独写一个函数

fireinice commented on src/models/morgan_chembert_model.py:
```diff
@@ -1,67 +1,80 @@
-        return result
+        out_embedding = weight_one * out_embedding_one + weight_two * out_embedding_two
+        x = self.mlp(out_embedding)
+        if self.loss_type == 'OR' and self.Ensemble_Learn:
```
这个变量为什么起了个这么奇怪的名字

fireinice commented on src/models/morgan_chembert_model.py:
```diff
@@ -1,67 +1,80 @@
+        out_embedding = weight_one * out_embedding_one + weight_two * out_embedding_two
+        x = self.mlp(out_embedding)
+        if self.loss_type == 'OR' and self.Ensemble_Learn:
+            predict = [classifier(x) for classifier in self.predict_layer]
```
这部分代码应该在父类中实现一致的逻辑，这里继承复用就可以了

fireinice commented on src/models/morgan_model.py:
```diff
@@ -13,50 +17,157 @@ def __init__(
         classify=True,
-        num_classes=52,
-        loss_type="OR",
+        num_classes=24,
```
这里有的是24,有的是52,建议都统一成2

fireinice reviewed (changes requested):
有一些代码可以单独写成一函数方便继承复用

fireinice commented on src/datamodule/baseline_datamodule.py:
```diff
@@ -1,64 +1,214 @@
-            bins=config.bins,
-        )
+    def calculate_weights(self, label_dict, dataset):
+        arr = []
```
这里的weights是基于每个item的，不是基于类的，如果有90万的条目，需要一个90万长度的list

fireinice reviewed (changes requested):
看一下评审意见

fireinice commented on src/datamodule/morgan_chembert_datamodule.py:
```diff
@@ -2,56 +2,181 @@
+                self.drug_featurizer_two,
+                self.target_featurizer,)
+
+    def calculate_weights(self, label_dict, dataset):
```
可以参考这个实现：
    def calculate_weights(self, label_dict, dataset):
        arr = []
        for _, _, label in dataset:
            count = label_dict[label.item()]
            if count == 0:
                weight = 0
            else:
                weight = len(dataset) / count
            arr.append(weight)
        return arr

xiangyuli-bio committed:
修改不平衡权重和代码模块化

1.baseline_datamodule.py 新增porcess_data 用于数据前期加载
2.dg_datamodule.py 新增build_weighted_sampler函数 用于返回sampler
3.morgan_chembert_datamodule.py 根据新增函数对应调整
4.morgan_model.py 新增classifier_forward函数：分类器
5.morgan_chembert_model.py 根据新增函数调整
